### PR TITLE
Update `redis` to `4.5.5`

### PR DESCRIPTION
**Pull request by sobolevn**
Closes https://github.com/python/typeshed/pull/10159

### CODE REVIEW

**Comment by sobolevn** [comment]
I had to add some `type: ignore[override]` comments, because this is how `redis` defines these methods. A bit messy, but nothing really problematic.

File: stubs/redis/redis/cluster.pyi
Comment (AlexWaygood) [inline stubs/redis/redis/cluster.pyi:1]: There's also a new `parse_cluster_myshardid` function in this module: https://github.com/redis/redis-py/compare/v4.5.4...v4.5.5#diff-a324b052f8b1189052b9606664591ba6bfe7669e971ddd027093413bc24bd372R116-R120

And a new `remap_host_port` method on the `NodesManager` class: https://github.com/redis/redis-py/compare/v4.5.4...v4.5.5#diff-a324b052f8b1189052b9606664591ba6bfe7669e971ddd027093413bc24bd372R1614-R1623

File: stubs/redis/redis/cluster.pyi [hunk 1]
```diff
@@ -149,6 +150,7 @@ class NodesManager:
         lock: Lock | None = None,
         dynamic_startup_nodes: bool = True,
         connection_pool_class: type[ConnectionPool] = ...,
+        address_remap: Callable[[str, int], tuple[str, int]] | None = None,
```
Comment (AlexWaygood) [inline stubs/redis/redis/cluster.pyi:153]: You also need to add the attribute annotation to the class: https://github.com/redis/redis-py/compare/v4.5.4...v4.5.5#diff-a324b052f8b1189052b9606664591ba6bfe7669e971ddd027093413bc24bd372R1300

File: stubs/redis/redis/commands/core.pyi
Comment (AlexWaygood) [inline stubs/redis/redis/commands/core.pyi:1]: Looks like the `ManagementsCommands` class in this module also has a new `client_no_touch` method: https://github.com/redis/redis-py/compare/v4.5.4...v4.5.5#diff-b6e2b30a96585d53406ae02c597de930c0a6524d78e408563f140e2c67866aacR764-R773

**Review by AlexWaygood — COMMENTED** [review]
Thanks! Looks good, but I think there's a few things missing. As well as my inline comments, there's also a new `cluster_myshardid` method added to the `ClusterManagementCommands` class in `redis/commands/cluster.py`: https://github.com/redis/redis-py/compare/v4.5.4...v4.5.5#diff-ab663f2062b059dda362204814b3195e6157efd254c879b3826c00f973ceed59R636-R643

**Review by AlexWaygood — APPROVED** [review]
LGTM!

**Pull request merged**